test/test_misc.f90

```diff
@@ -334,8 +337,78 @@ program test_misc
   call shrink(array_of_reals, 2_i64)
   call test_array(itest)%assert(array_of_reals, [1, 2]*1.0_r64)
   
+  ! Hilbert transform tests (H)
+  ! fx1 -> function 1, fx2 -> function 2
+  ! hfx1_even stores hilbert transform calculated for fx1, and for even number
+  ! of points
+  xmin = -30.0
+  xmax = 30.0
+  n_even = 4000
+  n_odd = 4001
+  ! ind_even are indices to compare in case of even number of points
+  ! ind_odd are indices to compare in case of odd number of points
+  allocate(ind_even(6),ind_odd(5))
+  ind_even = [801, 1201, 1601, 2001, 2401, 2801]
+  ind_odd = [889, 1333, 1777, 2221, 2665]
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: f(x) = 1/(1 + x^2), even points")
+  allocate(x_even(n_even), hfx1_even(n_even))
+  call linspace(x_even, xmin, xmax, n_even)
+  call Hilbert_transform(fx1(x_even), hfx1_even)
+  call test_array(itest)%assert(hfx1_even(ind_even), hfx1(x_even(ind_even)), &
+       tol = 2e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: f(x) = sin(x)/(1 + x^2), even points")
+  allocate(hfx2_even(n_even))
+  call Hilbert_transform(fx2(x_even), hfx2_even)
+  call test_array(itest)%assert(hfx2_even(ind_even), hfx2(x_even(ind_even)), &
+       tol = 1e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: f(x) = 1/(1 + x^2), odd points")
+  allocate(x_odd(n_odd), hfx1_odd(n_odd))
+  call linspace(x_odd, xmin, xmax, n_odd)
+  call Hilbert_transform(fx1(x_odd), hfx1_odd)
+  call test_array(itest)%assert(hfx1_odd(ind_odd), hfx1(x_odd(ind_odd)), &
+       tol = 4e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: f(x) = sin(x)/(1 + x^2), odd points")
+  allocate(hfx2_odd(n_odd))
+  call Hilbert_transform(fx2(x_odd), hfx2_odd)
+  call test_array(itest)%assert(hfx2_odd(ind_odd), hfx2(x_odd(ind_odd)), &
+       tol = 1e-5_r64)
+
   tests_all = testify(test_array)
   call tests_all%report
   
   if(tests_all%get_status() .eqv. .false.) error stop -1
+
+  contains
+     ! reference functions for the Hilbert transform test
+     ! fx1 is an even function
+     pure elemental real(r64) function fx1(x)
+        real(r64), intent(in) :: x
+        fx1 = 1/(1.0_r64 + x**2)
+     end function fx1
+
+     ! Hfx1 is actual hilbert transform of fx1, is an odd function
+     pure elemental real(r64) function hfx1(x)
+        real(r64), intent(in) :: x
+        hfx1 = x/(1.0_r64 + x**2)
+     end function hfx1
+
+     ! fx2 is an odd function
+     pure elemental real(r64) function fx2(x)
+        real(r64), intent(in) :: x
+        fx2 = sin(x)/(1.0_r64 + x**2)
+     end function fx2
+
+     ! Hfx2 is actual hilbert transform of fx2, is an even function
+     pure elemental real(r64) function hfx2(x)
+        real(r64), intent(in) :: x
+        hfx2 = (1/exp(1.0_r64) - cos(x))/(1.0_r64 + x**2)
```

Comment:
Did you actually check if `1/exp(1.0_r64)` has the same precision as `1.0_r64/exp(1.0_r64)`? Why not simply write `exp(-1.0_r64)`?